rules/aip0233/http_body_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0233
+
+import (
+	"testing"
+
+	"github.com/googleapis/api-linter/rules/internal/testutils"
+)
+
+func TestHttpBody(t *testing.T) {
+	tests := []struct {
+		testName   string
+		body       string
+		methodName string
+		problems   testutils.Problems
+	}{
+		{"Valid", "*", "BatchCreateBooks", nil},
+		{"Invalid", "", "BatchCreateBooks", testutils.Problems{{Message: "*"}}},
+		{"Irrelevant", "", "AcquireBook", nil},
+	}
+
+	for _, test := range tests {
+		t.Run(test.testName, func(t *testing.T) {
+			template := `import "google/api/annotations.proto";
+
+service BookService {
+	rpc {{.MethodName}}({{.MethodName}}Request) returns ({{.MethodName}}Response) {
+		option (google.api.http) = {
+			post: "/v1/{parent=publishers/*}/books:batchCreate"
+			body: "{{.Body}}"
+		};
+	}
+}
+
+message {{.MethodName}}Request {}
+
+message {{.MethodName}}Response{}
+`
+			file := testutils.ParseProto3Tmpl(t, template,
+				struct {
```

Comment:
Instead of passing a custom `struct` here, export the `MethodName` and `Body` properties on the test struct itself, and send `test`.

Applies throughout the PR.

Reply:
Done.